modular_skyrat/modules/conveyor_sorter/code/conveyor_sorter.dm

```diff
@@ -145,3 +146,71 @@
 		"conveysorter",
 	)
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 2500)
+
+/obj/item/conveyor_sorter/improved
+	name = "improved conveyor sorter lister"
+	desc = "A tool that is used to not only create the conveyor sorters, but give lists to the conveyor sorters. This one has an increased capacity of eight."
+	icon_state = "lister_improved"
+
+/obj/item/conveyor_sorter/improved/examine(mob/user)
+	. = ..()
+	. += span_notice("This version of the sorter allows up to eight sorters to be placed.")
+
+/obj/item/conveyor_sorter/improved/attack_self(mob/user, modifiers)
+	if(length(spawned_sorters) >= 8)
+		to_chat(user, span_warning("You may only have eight spawned conveyor sorters!"))
+		return
+	var/obj/effect/decal/cleanable/conveyor_sorter/improved/new_cs = new /obj/effect/decal/cleanable/conveyor_sorter/improved(get_turf(src))
+	new_cs.parent_item = src
+	new_cs.sorting_list = src.current_sort
+	spawned_sorters += new_cs
+
+/obj/effect/decal/cleanable/conveyor_sorter/improved
+	name = "improved conveyor sorter"
+	desc = "A mark that will sort items out based on what they are. This one can sort in multiple directions."
+	icon = 'modular_skyrat/modules/conveyor_sorter/icons/conveyor_sorter.dmi'
+	icon_state = "sorter_improved"
+	light_range = 3
+	light_color = COLOR_BLUE_LIGHT
+
+/obj/effect/decal/cleanable/conveyor_sorter/improved/attack_hand(mob/living/user, list/modifiers)
+	var/user_choice_improved = tgui_input_list(user, "Choose which direction to sort to!", "Direction choice", list("North", "East", "South", "West", "North-East", "North-West", "South-East", "South-West"))
```

Comment:
I believe there is already a global list for cardinals + ordinals

Reply:
is there? what is it named and i can set it as that

Reply:
I believe it is GLOB.alldirs

Reply:
i did try that but it resulted in breaking it entirely, well not entirely but made it a lot more confusing as to what direction the user was picking 
![image](https://user-images.githubusercontent.com/2568378/215297455-d986df6f-8dbf-48b0-b15e-30c3319a86b0.png)

Reply:
I'd recommend ditching the dashes for `North-East` and so on (so they look like `NorthEast`/`Northeast`, and use `text2dir` instead of your switch.

Reply:
> I'd recommend ditching the dashes for `North-East` and so on (so they look like `NorthEast`/`Northeast`, and use `text2dir` instead of your switch.
> 
> Or you can use `replacetext` to replace the hyphen when passing to `text2dir`. Up to you.

Done, and works nicely